spec/models/culinary_recipe_food_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe RecipeFood, type: :model do
+  it 'has a valid factory' do
+    culinary_recipe_food = FactoryBot.build(:culinary_recipe_food)
+    expect(culinary_recipe_food).to be_valid
+  end
+
+  it 'is not valid without a quantity' do
+    culinary_recipe_food = FactoryBot.build(:culinary_recipe_food, quantity: nil)
+    expect(culinary_recipe_food).to_not be_valid
+  end
+
+  it 'requires quantity to be a non-negative integer' do
+    culinary_recipe_food = FactoryBot.build(:culinary_recipe_food, quantity: -1)
+    expect(culinary_recipe_food).to_not be_valid
+
+    culinary_recipe_food = FactoryBot.build(:culinary_recipe_food, quantity: 5) # Adjust the value as needed
+    expect(culinary_recipe_food).to be_valid
+  end
+
+  it 'belongs to a recipe' do
+    recipe_association = RecipeFood.reflect_on_association(:recipe)
+    expect(recipe_association.macro).to eq(:belongs_to)
```

Comment:
- You have done well so far but some of your tests are currently failing, please ensure that all tests pass 👍 👍
<img width="637" alt="ywy3wyt" src="https://github.com/Anand8317/recipe/assets/73607512/58311897-75fe-4509-8782-2fb39a5db055">
 
- Also according to the project requirements, you are required to write integration tests for your application, please add integration tests to your app 👍 👍

Reply:
- [ ] [REQUIRED] Kindly ensure that the failed tests pass as recommended by the previous reviewer. This is a hard requirement. 
- [ ] Kindly add integration tests to your app as stated by the project requirement.

<img width="447" alt="Screenshot 2024-02-18 at 01 05 54" src="https://github.com/Anand8317/recipe/assets/58692590/96cfb949-3ca8-45b6-aa9b-0868a8540114">